### PR TITLE
chore(deps): Upgrade NX to v16.8.x

### PR DESCRIPTION
**Pull request by jakemhiller**
### Overview

Upgrade NX to v16.8.0

* Moves all dev dependencies up to the root package.json
* Updates various internal npm dependencies
* Shouldn't affect the build output of the packages (tested this with `npx npm-diff` against the alphas)

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: [ABC-123]
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change
- [ ] This PR includes testing instructions tests for the code change
- [ ] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### Testing Instructions

[Don't make me tap the sign.](https://i.imgur.com/sy93D9I.png)

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](http://www.google.fr/ 'Named link title')  | [Monolith Env](http://www.google.fr/ 'Named link title') |
| Portal       | [Portal Link](http://www.google.fr/ 'Named link title')  | [Portal Env](http://www.google.fr/ 'Named link title')   |
| Another Repo | [Another Link](http://www.google.fr/ 'Named link title') | [Another Env](http://www.google.fr/ 'Named link title')  |

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>eslint-plugin-gamut@2.0.4-alpha.472d6c.0<br/>@codecademy/gamut@55.3.2-alpha.472d6c.0<br/>@codecademy/gamut-icons@9.4.1-alpha.472d6c.0<br/>@codecademy/gamut-illustrations@0.39.1-alpha.472d6c.0<br/>@codecademy/gamut-kit@0.6.342-alpha.472d6c.0<br/>@codecademy/gamut-patterns@0.9.8-alpha.472d6c.0<br/>@codecademy/gamut-styles@16.1.3-alpha.472d6c.0<br/>@codecademy/gamut-tests@5.0.7-alpha.472d6c.0<br/>@codecademy/macros@3.0.3-alpha.472d6c.0<br/>@codecademy/styleguide@66.1.27-alpha.472d6c.0<br/>@codecademy/variance@0.21.1-alpha.472d6c.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>eslint-plugin-gamut@2.0.4-alpha.0f0cb9.0<br/>@codecademy/gamut@55.3.2-alpha.0f0cb9.0<br/>@codecademy/gamut-icons@9.4.1-alpha.0f0cb9.0<br/>@codecademy/gamut-illustrations@0.39.1-alpha.0f0cb9.0<br/>@codecademy/gamut-kit@0.6.342-alpha.0f0cb9.0<br/>@codecademy/gamut-patterns@0.9.8-alpha.0f0cb9.0<br/>@codecademy/gamut-styles@16.1.3-alpha.0f0cb9.0<br/>@codecademy/gamut-tests@5.0.7-alpha.0f0cb9.0<br/>@codecademy/macros@3.0.3-alpha.0f0cb9.0<br/>@codecademy/styleguide@66.1.27-alpha.0f0cb9.0<br/>@codecademy/variance@0.21.1-alpha.0f0cb9.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>eslint-plugin-gamut@2.0.4-alpha.187436.0<br/>@codecademy/gamut@55.4.1-alpha.187436.0<br/>@codecademy/gamut-icons@9.4.1-alpha.187436.0<br/>@codecademy/gamut-illustrations@0.39.1-alpha.187436.0<br/>@codecademy/gamut-kit@0.6.343-alpha.187436.0<br/>@codecademy/gamut-patterns@0.9.8-alpha.187436.0<br/>@codecademy/gamut-styles@16.1.3-alpha.187436.0<br/>@codecademy/gamut-tests@5.0.7-alpha.187436.0<br/>@codecademy/macros@3.0.3-alpha.187436.0<br/>@codecademy/styleguide@66.2.1-alpha.187436.0<br/>@codecademy/variance@0.21.1-alpha.187436.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>eslint-plugin-gamut@2.0.4-alpha.2b1053.0<br/>@codecademy/gamut@55.4.1-alpha.2b1053.0<br/>@codecademy/gamut-icons@9.4.1-alpha.2b1053.0<br/>@codecademy/gamut-illustrations@0.39.1-alpha.2b1053.0<br/>@codecademy/gamut-kit@0.6.343-alpha.2b1053.0<br/>@codecademy/gamut-patterns@0.9.8-alpha.2b1053.0<br/>@codecademy/gamut-styles@16.1.3-alpha.2b1053.0<br/>@codecademy/gamut-tests@5.0.7-alpha.2b1053.0<br/>@codecademy/macros@3.0.3-alpha.2b1053.0<br/>@codecademy/styleguide@66.2.1-alpha.2b1053.0<br/>@codecademy/variance@0.21.1-alpha.2b1053.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>eslint-plugin-gamut@2.0.4-alpha.d60f39.0<br/>@codecademy/gamut@55.4.1-alpha.d60f39.0<br/>@codecademy/gamut-icons@9.4.1-alpha.d60f39.0<br/>@codecademy/gamut-illustrations@0.39.1-alpha.d60f39.0<br/>@codecademy/gamut-kit@0.6.343-alpha.d60f39.0<br/>@codecademy/gamut-patterns@0.9.8-alpha.d60f39.0<br/>@codecademy/gamut-styles@16.1.3-alpha.d60f39.0<br/>@codecademy/gamut-tests@5.0.7-alpha.d60f39.0<br/>@codecademy/macros@3.0.3-alpha.d60f39.0<br/>@codecademy/styleguide@66.2.1-alpha.d60f39.0<br/>@codecademy/variance@0.21.1-alpha.d60f39.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6501e4e177bf81193744c751--gamut-preview.netlify.app>https://6501e4e177bf81193744c751--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6501e4e177bf81193744c751>Deploy Logs</a></p>

File: jest.preset.js [hunk 1]
```diff
@@ -1,5 +1,14 @@
...
+const targetProject = kebabCase(
+  process.env.NX_TASK_TARGET_PROJECT || 'unknown'
+);
 
 module.exports = {
   ...nxPreset,
+  coverageReporters: ['json', 'text', 'clover'],
+  reporters: process.env.CI
+    ? ['default', ['jest-junit', { outputName: `${targetProject}-report.xml` }]]
+    : ['default'],
```
Comment (jakemhiller) [inline jest.preset.js:13]: moved these up into the preset to avoid some warnings from jest

File: packages/gamut/package.json [hunk 2]
```diff
@@ -56,15 +56,6 @@
...
   "license": "MIT",
-  "devDependencies": {
-    "@codecademy/gamut-tests": "5.0.6",
-    "@emotion/jest": "^11.3.0",
-    "@testing-library/dom": "^8.11.1",
-    "@testing-library/jest-dom": "^5.16.5",
-    "@testing-library/react": "^12.1.2",
-    "@testing-library/user-event": "^13.5.0",
-    "onchange": "^7.0.2"
-  },
```
Comment (jakemhiller) [inline packages/gamut/package.json:67]: dev dependencies are now ALL handled by the root package.json

File: .lintstagedrc.mjs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  [`*`]: (allChanges) => {
+    const commands = [];
+
+    // if (micromatch.some(allChanges, '**/package.json')) {
+    //   commands.push(`yarn syncpack format`);
+    // }
+
+    // if (micromatch.some(allChanges, 'yarn.lock')) {
+    //   commands.push(`npx yarn-deduplicate`);
+    // }
```
Comment (dreamwasp) [inline .lintstagedrc.mjs:14]: do we need these comments?

File: .syncpackrc.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    'scripts',
+  ],
+  sortFirst: ['name', 'description', 'version', 'author'],
+  // versionGroups: [
+  //   {
+  //     packages: ['**'],
+  //     dependencies: ['**'],
+  //     snapTo: ['gamut-repo'],
+  //   },
+  // ],
```
Comment (dreamwasp) [inline .syncpackrc.js:24]: ^^

**Review by dreamwasp — APPROVED** [review]
love it, lets gooooo

File: .lintstagedrc.mjs (hunk 3 above)
Comment (jakemhiller) [reply .lintstagedrc.mjs:14]: i'm going to enable these in a couple of follow-up PRs

Comment (dreamwasp) [reply .lintstagedrc.mjs:14]: perf

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>eslint-plugin-gamut@2.0.4-alpha.5bf9ff.0<br/>@codecademy/gamut@55.4.1-alpha.5bf9ff.0<br/>@codecademy/gamut-icons@9.4.1-alpha.5bf9ff.0<br/>@codecademy/gamut-illustrations@0.39.1-alpha.5bf9ff.0<br/>@codecademy/gamut-kit@0.6.343-alpha.5bf9ff.0<br/>@codecademy/gamut-patterns@0.9.8-alpha.5bf9ff.0<br/>@codecademy/gamut-styles@16.1.3-alpha.5bf9ff.0<br/>@codecademy/gamut-tests@5.0.7-alpha.5bf9ff.0<br/>@codecademy/macros@3.0.3-alpha.5bf9ff.0<br/>@codecademy/styleguide@66.2.1-alpha.5bf9ff.0<br/>@codecademy/variance@0.21.1-alpha.5bf9ff.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>eslint-plugin-gamut@2.0.4-alpha.9d771e.0<br/>@codecademy/gamut@55.4.1-alpha.9d771e.0<br/>@codecademy/gamut-icons@9.4.1-alpha.9d771e.0<br/>@codecademy/gamut-illustrations@0.39.1-alpha.9d771e.0<br/>@codecademy/gamut-kit@0.6.343-alpha.9d771e.0<br/>@codecademy/gamut-patterns@0.9.8-alpha.9d771e.0<br/>@codecademy/gamut-styles@16.1.3-alpha.9d771e.0<br/>@codecademy/gamut-tests@5.0.7-alpha.9d771e.0<br/>@codecademy/macros@3.0.3-alpha.9d771e.0<br/>@codecademy/styleguide@66.2.1-alpha.9d771e.0<br/>@codecademy/variance@0.21.1-alpha.9d771e.0<br/></pre>

**Pull request merged**